CONTRIBUTING.md

```diff
@@ -0,0 +1,132 @@
+# Contributing to the documentation
+
+## Repository structure
+
+Briefly, this is what you need to know about the documentation repository structure before contributing:
+
+* The documentation is module-based, which means most sections are individual modules. These modules are available in the */docs/topics/modules/* folder.
+* The modules are linked together to form a chapter, document, or guide using assemblies. These assemblies are asciidoc files in the */docs/topics/* folder. For example, the */docs/topics/getting-started-guide.adoc* file is the main assembly for the Getting Started Guide and links to other, smaller assemblies for parts or chapters within the guide.
+* Module files naming conventions closely follow the title of the corresponding section. For example, a section in the Getting Started document with the title *Working with pipelines* is named **working_with_pipelines.adoc** in the *modules* folder. In some cases, this is not a direct match, so to confirm the name of the asciidoc file for a section, click the section title and the ID after the *#* in the URL is the name of the module file. For example, a URL such as https://docs.openshift.io/getting-started-guide.html#viewing_build_pipeline_oso has a slightly different title but the ID after the hash symbol (**viewing_build_pipeline_oso**) is the name of the file.
+
+## Conventions and style guidelines
+
+To get started quickly, we recommend reviewing the following conventions that are relevant for this documentation suite:
+
+**Writing conventions**
+
+* For instructions, state the location, then action and subject. For example:
+  * Incorrect: *Click OK in the XYZ tab to continue.*
+  * Correct: *In the XYZ tab, click OK to continue.*
+  * Use gerunds in titles when possible. For example:
+    * Incorrect: *1.1 Create a new project*
+    * Correct: *1.1 Creating a new project*
+* All titles should be in sentence case. Exceptions for words that are proper nouns. For example:
+  * Incorrect: *1.2 Configuring Your Che Workspace*
+  * Correct: *1.2 Configuring your Che workspace*
+* URLs must be bared in the text, not hidden. For example:
+  * Incorrect: *Navigate to the [Red Hat website](www.redhat.com).* 
+  * Correct: *Navigate to (www.redhat.com).*
+  
+**Markup conventions**  
+
+* TBD 
+  
```

Comment:
Suggestion: How about starting with the references and then the other conventions followed in this repo? I mean IBM and later CCS have writing conventions in them too, so maybe we could keep it as just one paragraph. 
A couple of things we need to add:
The usage of capital letters to refer to UI elements. Example, A space can also include areas. v/s Click **Areas**.
TBD: When we share Screenshots across the guides can we agree to follow a naming convention of appending a '_c'  to the name, example, xyz_c.png. The concern being, if for some reason we update a screenshot in one guide, it might loose relevance in the other guide. Basically a way of letting the writers know that changing a screenshot could have an effect in another guide. wdyt?

Reply:
Just noticed, should we not be using asciidoc for both the docs instead of markdown? Probably something for later.

Reply:
Preeti, the reason for the shorter conventions for our docs being first is for users who don't really care about knowing the larger conventions (and fair enough) to read lengthy and detailed docs about them, so a quick list of the ones you need to know for this doc is first. The other stuff we (the reviewers can fix up ourselves since we do know them because this is what we do).

Reply:
@Preeticp For the screenshot naming, could you elaborate a bit more? Are you suggesting that they should all be named with a _c at the end, or just that if we update, we change this appended text? A naming convention that requires a lot of changing of the name doesn't sound great for maintenance. We can probably work out a script that checks if image.png is not being used at all in any doc and manually get rid of it, but otherwise, I'm not sure I understand the benefit you're proposing here.

Reply:
@Preeticp re: using adoc instead of MD for these readme style documents, I don't know if they will build the same way, but if there's no different, I'm fine with either. There's no real difference since both are reasonably simple syntaxes.

Reply:
Re: "The usage of capital letters to refer to UI elements. Example, A space can also include areas. v/s Click Areas." - Now that I think about this, it seems reasonable to expect that users will refer to UI elements as they are on the UI. For example, if a button says "ok", they will refer to it as the ok button rather than the OK button so as long as the capitalization (or lack thereof) matches the UI, it should be fine, IMO. WDYT?

Reply:
>@Preeticp For the screenshot naming, could you elaborate a bit more? Are you suggesting that they should all be named with a _c at the end, or just that if we update, we change this appended text? 

No, only those screenshots which would be common to both the User guide and GSG should have that tag. 

> A naming convention that requires a lot of changing of the name doesn't sound great for maintenance. We can probably work out a script that checks if image.png is not being used at all in any doc and manually get rid of it, but otherwise, I'm not sure I understand the benefit you're proposing here.

@mishaone my suggestion is, the moment one of us comes across an image we want to reuse for another guide we mark it as image_c.png. So that whenever anyone updates these images for their PR, they could leave a comment in the PR saying this changes a commonly used screenshot. (So that the writer is aware it impacts another guide(section) and informs others too of the change.) It is just a suggestion, I am open to other ways of resolving this situation.
The other alternative I can think of is to have separate screenshots for both the guides, which means more maintenance.

Reply:
> Re: "The usage of capital letters to refer to UI elements. Example, A space can also include areas. v/s Click Areas." - Now that I think about this, it seems reasonable to expect that users will refer to UI elements as they are on the UI. For example, if a button says "ok", they will refer to it as the ok button rather than the OK button so as long as the capitalization (or lack thereof) matches the UI, it should be fine, IMO. WDYT?

Phew, this one required quite a bit of searching, and precisely the reason we should add this in the conventions. We have had this conversation before, happy to follow what ever we decide on this: https://github.com/fabric8io/fabric8-online-docs/pull/114#issuecomment-348924579
As a best practice, whenever we have such discussions and reach a decision we should update this doc.

Reply:
@mishaone @Preeticp Ad capitals, I believe the result of the latest discussion was:

Use IBM-recommended caps (first letter cap, all else small) when not describing specific UI elements; file bugs against UI when UI elements are wrong™.

Reply:
@Preeticp I'm happy to go with the consensus on this, but from what I can tell, the GS doc has very specific screenshots (red circles pointing to important things and also showing specific file names and things for the quickstart). This is exactly why we have separate modules for the UG and GS doc. I would suspect that there are actually very few instances of an image being used in both docs. I can look into this and compile a quick list so we can compare if you like. In the meantime if you have a look through the images in the GS doc build, you'll probably see what I mean: lots of specific names and red circles, etc.

Reply:
Also, the IBM recommendation sounds good. Do we want to explicitly state this?